py/src/braintrust/wrappers/openai.py

```diff
@@ -212,12 +212,16 @@ def on_span_start(self, span: tracing.Span[tracing.SpanData]) -> None:
             parent = self._spans[span.parent_id]
         else:
             parent = self._spans[span.trace_id]
-        self._spans[span.span_id] = parent.start_span(
+        created_span = parent.start_span(
             id=span.span_id,
             name=_span_name(span),
             type=_span_type(span),
             start_time=_timestamp_from_maybe_iso(span.started_at),
         )
+        self._spans[span.span_id] = created_span
+
+        # Set the span as current so current_span() calls will return it
+        created_span.set_current()
```

Comment:
Cool. But now I'm realizing we may not have an analogous `unset_current` method that resets the context token created by `set_current`. And that unwinding only happens in the `__exit__` method. Maybe we should add that to the `Span` interface/implementation and use it here too?